src/main/scala/uk/gov/hmrc/govukfrontend/views/viewmodels/CommonJsonFormats.scala

```diff
@@ -56,4 +57,60 @@ object CommonJsonFormats {
     override def writes(o: Option[Html]): JsObject =
       o.map(o => Json.obj("conditional" -> Json.obj("html" -> o.body))).getOrElse(Json.obj())
   }
+
+  val attributesReads: Reads[Map[String, String]] = new Reads[Map[String, String]] {
+    override def reads(json: JsValue): JsResult[Map[String, String]] = {
+      val keyValueTuples = json.as[JsObject].keys.map { key =>
+        val maybeValue: Option[String] = (json \ key).asOpt[String].orElse {
+          (json \ key).asOpt[Int].map(_.toString).orElse {
+            (json \ key).asOpt[Boolean].map(_.toString)
+          }
+        }
+        maybeValue.map(v => (key, v))
+      }
+      JsSuccess(keyValueTuples.flatten.toMap)
+    }
+  }
+
+  val forgivingOptStringReads: Reads[Option[String]] = new Reads[Option[String]] {
+    override def reads(json: JsValue): JsResult[Option[String]] = {
+      val maybeString = json.asOpt[String].orElse {
+        json.asOpt[Int].map(_.toString).orElse {
+          json.asOpt[Boolean].map(_.toString)
+        }
+      }
+      JsSuccess(maybeString)
+    }
+  }
+
+  val forgivingStringReads: Reads[String] = new Reads[String] {
+    override def reads(json: JsValue): JsResult[String] = {
+      val maybeString = json.asOpt[String].orElse {
```

Comment:
There is a little duplication of the code from line 77 - could it be factored out?

Reply:
Cleaned up and pushed to branch.